### ISSUE TITLE
Defining a Core API

### ISSUE THREAD

**Issue by stschiff** [sidora-tools/sidora.core#1]
#' would return a table either from Cache as tibble or as tbl() dbplyer object
    
    get_table(db_con, entityType, fromCache=T, toCache=T)
    ...

   filter_table(input, filter_spec)

    join_tables(table_source, ...) -> would give error if you lack an intermediate table
examples:
    
    join_tables(sites, individuals, samples)
    error: join_tables(individuals, samples, libraries)

**Comment by nevrome** [comment]
I have serious difficulties with this dbplyr interface. It's very limited compared to standard dplyr, because it only allows

**Comment by nevrome** [comment]
1b84c39318cffdba96e9366fcedc12116f04479a to 4085731032b59d94bfc3ca10bc102dcd2c50fed6 show my journey to implement a simple filter function that works both with downloaded data as well as  with the dbplyr/MySQL connection interface. Unfortunately by far not all dplyr syntax can be used for the latter and I found it incredibly irritating to try all these things and see them fail one after another because of the fragile nature of this interface. 

If you, @stschiff and @jfy133, don't spot some obvious flaws in the way I approached this (maybe it's incredibly simple with some basic rules in mind?) then I would suggest to only use this high-level prefiltering for very basic operations.
